### ISSUE TITLE
Trouble creating a DuckDB table with `dbplyr::sql`

### ISSUE THREAD

**Issue by chriscardillo** [duckdb/duckdb#6506]
### What happens?

When attempting to use dbplyr::sql to create a dplyr::tbl using a duckdb connection, I receive the following error:

```
Error: Parser Error: syntax error at or near "SELECT"
LINE 2: FROM SELECT * FROM items AS q01
```

Originally raised the issue in the dbplyr repo, but they asked me to ask here instead: https://github.com/tidyverse/dbplyr/issues/1173

### To Reproduce

```r
# Creating connection and data
con <- DBI::dbConnect(duckdb::duckdb(), dbdir = ":memory:")
DBI::dbExecute(con, "CREATE TABLE items(item VARCHAR, value DECIMAL(10,2), count INTEGER)")
DBI::dbExecute(con, "INSERT INTO items VALUES ('jeans', 20.0, 1), ('hammer', 42.2, 2)")

# Confirming data is in the db
DBI::dbGetQuery(con, "SELECT * FROM items")

# Works
dplyr::tbl(con, "items")

# Does not work
dplyr::tbl(con, dbplyr::sql("SELECT * FROM items"))
```

In other languages, this is not an issue, and it's worth noting that this issue is resolved by wrapping the SQL in parentheses (e.g. `dplyr::tbl(con, dbplyr::sql("(SELECT * FROM items)"))`.

### OS:

x86

### DuckDB Version:

0.7.0

### DuckDB Client:

R

### Full Name:

Chris Cardillo

### Affiliation:

maintainer of the dbooper R package (chriscardillo/dbcooper)

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by rsund** [comment]
Thank you for the report.

Currently DuckDB's `tbl`-method is tailored (different from the `dbplyr`'s generic), because it checks if the string provided to it is corresponding to the database table and if not removes the quotes that `dbplyr` will add there otherwise in order to allow replacement scans and DuckDB functions to work directly (such as `dplyr::tbl(con, "read_parquet('test.parquet')")` ).

Could you please add `debug=TRUE` to your connection (i.e. use  `con <- DBI::dbConnect(duckdb::duckdb(), debug=TRUE)` ) and run the repro again and with the "parentheses added" version. That will show where the problem occurs in terms of generated sql.

Is there some specific reason that you could't use parenthesis there (wouldn't that work with `dbplyr`'s generic and therefore with other backends too) or create a view or temporary table before using `tbl`?

**Comment by chriscardillo** [comment]
Ran without parenthesis:

```
Q SELECT 1
Q SELECT * FROM "SELECT * FROM items" WHERE FALSE
Q SELECT *
FROM SELECT * FROM items AS q01
WHERE (0 = 1)
```

Ran with parentheses:

```
Q SELECT 1
Q SELECT * FROM "(SELECT * FROM items)" WHERE FALSE
Q SELECT *
FROM (SELECT * FROM items) AS q02
WHERE (0 = 1)
Q SELECT *
FROM (SELECT * FROM items)
LIMIT 11
```

It's the last `FROM` where the issue is occurring.

> Is there some specific reason that you could't use parenthesis there.

There is no reason I cannot use parentheses there, and to alleviate the issue I did just use the parentheses there, but because this behavior was unexpected, relative to running the same statement (no parentheses) with three other SQL dialects (Postgres, SQLite, Snowflake), I raised it as a bug. I just wanted to bring the behavior to your attention, should it be something you would want to remediate.

I will still be a big fan of DuckDB either way :grin:
